zebra-consensus/src/transaction.rs

```diff
@@ -473,7 +547,17 @@ where
                         ),
                         legacy_sigop_count,
                     )?;
-                    Response::Mempool { transaction }
+
+                    if let Some(mut mempool) = mempool {
+                        if !transaction.transaction.transaction.outputs().is_empty() {
+                            tokio::spawn(async move {
+                                tokio::time::sleep(POLL_MEMPOOL_DELAY).await;
+                                mempool.ready().await.expect("mempool poll_ready() method should not return an error");
+                            });
+                        }
+                    }
```

Comment:
This should be moved to the mempool's `Downloads` component, see [this comment](https://github.com/ZcashFoundation/zebra/pull/8901/files#r1794389135).